com.unity.multiplayer.mlapi/Tests/Runtime/RPCQueueTests.cs

```diff
@@ -0,0 +1,127 @@
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+using MLAPI.SceneManagement;
+using MLAPI.Transports.UNET;
+
+namespace MLAPI.RuntimeTests
+{
+    /// <summary>
+    /// The RPC Queue unit test to validate:
+    /// - Sending and Receiving pipeline to validate that both sending and receiving pipelines are functioning properly.
+    /// - Usage of the ServerRpcParams.Send.UpdateStage and ClientRpcParams.Send.UpdateStage functionality.
+    /// - Rpcs receive will be invoked at the appropriate NetworkUpdateStage.
+    /// Requires: RpcPipelineTestComponent
+    /// </summary>
+    public class RpcQueueTests
+    {
+        private NetworkingManager m_NetMan;
+
+#if UNITY_EDITOR
+        [UnitySetUp]
+        public IEnumerator SetUp()
+        {
+            yield return new EnterPlayMode();
+        }
+#endif
+
+        /// <summary>
+        /// Tests the egress and ingress RPC queue functionality
+        /// ** This does not include any of the MLAPI to Transport code **
+        /// </summary>
+        /// <returns>IEnumerator</returns>
+        [UnityTest]
+        public IEnumerator RpcQueueUnitTest()
+        {
+            var netManObject = new GameObject();
+            m_NetMan = netManObject.AddComponent<NetworkingManager>();
+            UnetTransport unetTransport = netManObject.AddComponent<UnetTransport>();
```

Comment:
Test

Reply:
FWIW I did this for my connect logic in my test cases.

`  GameObject o = new GameObject();
        NetworkingManager nm = (NetworkingManager)o.AddComponent(typeof(NetworkingManager));
        nm.SetSingleton();
        nm.NetworkConfig = new NetworkConfig();
        UnetTransport ut = (UnetTransport)o.AddComponent(typeof(UnetTransport));

        ut.ServerListenPort = 7777;
        nm.NetworkConfig.NetworkTransport = ut;

        try
        {
            nm.StartServer();
        }
        catch (InvalidChannelException e)
        {
            Assert.Fail("The server wouldn't start");
        }

        nm.StopServer();
        nm.Shutdown();
`